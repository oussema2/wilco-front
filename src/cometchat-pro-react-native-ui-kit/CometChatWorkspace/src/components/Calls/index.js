export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
