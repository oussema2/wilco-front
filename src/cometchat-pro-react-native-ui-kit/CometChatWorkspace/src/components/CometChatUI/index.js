/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
