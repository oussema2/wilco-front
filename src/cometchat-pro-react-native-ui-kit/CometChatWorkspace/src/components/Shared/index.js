export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
