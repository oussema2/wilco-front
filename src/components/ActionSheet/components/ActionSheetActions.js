import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
