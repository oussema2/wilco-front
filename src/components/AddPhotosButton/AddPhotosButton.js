import React from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react-lite';
import { TertiaryButton } from '../TertiaryButton';
import { photoVideo } from '../../assets/icons';
import PlaceholderButton from '../PlaceholderButton/PlaceholderButton';
import { useAddPhotosButton } from '../../hooks/useAddPhotosButton';

const AddPhotosButton = ( {
	testID, title, onPhotosSelected, selectedPhotos, maxPhotosAllowed, showPlaceholder
} ) => {
	const handleGalleryPermission = useAddPhotosButton(
		onPhotosSelected,
		maxPhotosAllowed,
		selectedPhotos
	);

	return (
		showPlaceholder
			? (
				<PlaceholderButton
					testID={testID}
					imageSource={photoVideo}
					onPress={handleGalleryPermission}
					title="Add photos"
				/>
			)
			: (
				<TertiaryButton
					testID={testID}
					title={title}
					size="small"
					onPress={handleGalleryPermission}
				/>
			)
	);
};

AddPhotosButton.propTypes = {
	testID: PropTypes.string,
	title: PropTypes.string,
	onPhotosSelected: PropTypes.func,
	selectedPhotos: PropTypes.any,
	maxPhotosAllowed: PropTypes.number,
	showPlaceholder: PropTypes.bool
};

AddPhotosButton.defaultProps = {
	testID: 'addPhotosButton-component',
	title: 'Add photo/s',
	onPhotosSelected: () => {},
	selectedPhotos: [],
	maxPhotosAllowed: 10,
	showPlaceholder: false
};

export default observer( AddPhotosButton );
