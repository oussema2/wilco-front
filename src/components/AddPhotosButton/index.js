export { default as AddPhotosButton } from './AddPhotosButton';
