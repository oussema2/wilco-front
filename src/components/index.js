export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
