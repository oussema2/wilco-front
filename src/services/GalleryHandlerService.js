import PermissionsHandlerService, {
	galleryPermission,
	permissionResults
} from './PermissionsHandlerService';
import AlertMessagesService from './AlertMessagesService';

let sharedInstance;

export default class GalleryHandlerService {
	constructor( {
		permissionHandlerService = PermissionsHandlerService.shared(),
		alertMessagesService = AlertMessagesService.shared()
	} = {} ) {
		this.permissionHandlerService = permissionHandlerService;
		this.alertMessagesService = alertMessagesService;
	}

	static shared = () => {
		if ( sharedInstance ) { return sharedInstance; }

		sharedInstance = new GalleryHandlerService();
		return sharedInstance;
	}

	async handleGalleryPermission( _callback ) {
		const requestResult = await this.permissionHandlerService.request( galleryPermission );

		const GRANTED_PERMISSION_RESULTS = [ permissionResults.granted, permissionResults.limited ];
		const permissionWasGranted = ( result ) => GRANTED_PERMISSION_RESULTS.includes( result );

		if ( permissionWasGranted( requestResult ) ) {
			_callback();
		} else {
			this._showAlertForOpenSettings();
		}
	}

	_showAlertForOpenSettings() {
		this.alertMessagesService.showConfirmationAlert( {
			title: '"Wilco" Would Like to Access Your Photos',
			message: 'This app requires access to the photo library.',
			cancelButton: { onPress: () => {} },
			confirmButton: {
				text: 'Open Settings',
				onPress: () => this.permissionHandlerService.openSettings()
			}
		} );
	}
}
