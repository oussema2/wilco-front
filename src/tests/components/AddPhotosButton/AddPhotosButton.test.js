import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { AddPhotosButton } from '../../../components/AddPhotosButton';
import GalleryHandlerService from '../../../services/GalleryHandlerService';

describe( '<AddPhotosButton />', () => {
	let component;
	const testID = 'add-photos-button-testID';
	const onPhotosSelected = jest.fn();
	const maxPhotosAllowed = 10;

	const setUp = ( props ) => {
		component = render(
			<AddPhotosButton
				testID={testID}
				onPhotosSelected={onPhotosSelected}
				maxPhotosAllowed={maxPhotosAllowed}
				{...props}
			/>
		);
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the AddPhotosButton component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();
			expect( component.queryByText( 'Add photo/s' ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with custom title', () => {
		it( 'renders the AddPhotosButton component with the given title', () => {
			setUp( { title: 'Custom title' } );
			expect( component.queryByText( 'Custom title' ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when user press the button', () => {
		it( 'calls to handleGalleryPermission', async () => {
			const galleryHandlerService = GalleryHandlerService.shared();
			const handleGalleryPermissionSpy = jest.spyOn( galleryHandlerService, 'handleGalleryPermission' );

			fireEvent.press( component.queryByTestId( testID ) );
			expect( handleGalleryPermissionSpy ).toHaveBeenCalled();
		} );
	} );
} );
