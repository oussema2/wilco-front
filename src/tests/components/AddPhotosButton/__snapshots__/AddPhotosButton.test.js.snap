// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPhotosButton /> with custom title renders the AddPhotosButton component with the given title 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="add-photos-button-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#5B5B5B",
          "borderRadius": 26,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 52,
          "justifyContent": "center",
          "padding": 8,
          "width": 164,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#5B5B5B",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 13,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        Custom title
      </Text>
    </View>
  </View>
</View>
`;

exports[`<AddPhotosButton /> with the default props renders the AddPhotosButton component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="add-photos-button-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#5B5B5B",
          "borderRadius": 26,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 52,
          "justifyContent": "center",
          "padding": 8,
          "width": 164,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#5B5B5B",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 13,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        Add photo/s
      </Text>
    </View>
  </View>
</View>
`;
